draft-ietf-jose-hpke-encrypt.md

```diff
@@ -253,15 +254,56 @@ Otherwise, the JWE Protected Header (and JWE Shared Unprotected Header) MUST NOT
 - JOSE Header parameter "psk_id" MAY be present.
 - JOSE Header parameter "ek" MUST be present and contain the base64url-encoded HPKE encapsulated key.
 - Recipient JWE Encrypted Key MUST be the ciphertext from HPKE Encryption.
-- The HPKE info parameter defaults to the empty string; mutually known private information MAY be used instead.
-- The HPKE AAD parameter MUST be set to the empty string.
+- The HPKE info parameter contains the encoding of the Recipient_structure, which is described in {{recipient_structure}}.
+- The HPKE AAD parameter defaults to the empty string; externally provided information MAY be used instead.
 - THE HPKE plaintext MUST be set to the CEK.
 
 The processing of "enc", "iv", "tag", "aad", and "ciphertext" is as already defined in {{RFC7516}}.
 Implementations process these parameters as defined in {{RFC7516}};
 no additional processing requirements are introduced by HPKE-based key encryption.
 
-## JSON Example
+## Recipient_structure {#recipient_structure}
+
+The `Recipient_structure` is a JSON object with the following members:
+
+- context (string): This member MUST contain the constant string value "JOSE HPKE Recipient".
+
+- next_layer_alg (string): Identifies the algorithm with which the HPKE-encrypted key MUST be used. Its value MUST match the "enc" (encryption algorithm) header parameter in the next lower JOSE layer.
+
+- recipient_protected_header (object): Contains the protected header parameters from the recipient structure, excluding the "ek" member.
+
+- recipient_extra_info (string): Contains additional context information that the application includes in the key derivation via the HPKE `info` parameter. If no additional context is provided, this value MUST be the empty string "".
+
+### Deterministic Serialization for HPKE `info`
+
+JSON texts that are semantically identical can serialize differently (e.g., member order, whitespace), which would lead to divergent `info` values and failed key agreement.
+
+To produce the HPKE `info` byte string from a `Recipient_structure`, implementations MUST use a deterministic JSON serialization to ensure both parties derive identical bytes regardless of JSON library differences (e.g., member order, whitespace, numeric formatting):
+
+1. Construct the `Recipient_structure` JSON object exactly as defined above.
+2. Serialize the object using the JSON Canonicalization Scheme (JCS) specified in {{RFC8785}}. Canonicalization fixes member ordering (lexicographic by Unicode code point), whitespace (none beyond JSON syntax), and value formatting.
```

Comment:
Why are we forcing implementers to bundle a canonicalization library in order to use encryption?

Lets just use the procedure described for JWK thumbprints instead, see https://datatracker.ietf.org/doc/html/rfc7638#section-3.2

Must simpler, and possibly a already present in code that is implemented this spec.